.gitlab-ci.yml

```diff
@@ -1,28 +1,27 @@
-image: python:3.7
+# GitLab Dependency Proxy to get rid of Docker hub pull limit issue
+# https://docs.gitlab.com/ee/user/packages/dependency_proxy/#authenticate-within-cicd
+image: ${CI_DEPENDENCY_PROXY_GROUP_IMAGE_PREFIX}/python:3.7
 
 # GitLab CI uses the services keyword to define what docker containers should be linked with our base image.
 # Since we are using docker as our image we'll be using docker:dind i.e. Docker-In-Docker-Image
 services:
-  - name: docker:19.03.12-dind
+  - name: ${CI_DEPENDENCY_PROXY_GROUP_IMAGE_PREFIX}/docker:dind
```

Comment:
Where do these images actually get used?  The pytest step is disabled, you don't use them in the build step, and there's no deploy step

Reply:
~I'll delete this global `services` and see what happens.~, I did and it worked, I removed this part from the code as it's not needed.
As for the build stage the `docker:dind` image is required because it starts up the docker daemon when it is brought up by a runner (I stole this line from [here](https://forum.gitlab.com/t/why-services-docker-dind-is-needed-while-already-having-image-docker/43534/2)).